modules/PersistentStore/main/kvsImpl.cpp

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 - 2022 Pionix GmbH and Contributors to EVerest
+
+#include "kvsImpl.hpp"
+
+#include <boost/filesystem.hpp>
+#include <boost/variant.hpp>
+
+namespace module {
+namespace main {
+
+void kvsImpl::init() {
+    // open and initialize database
+    boost::filesystem::path sqlite_db_path =
+        boost::filesystem::absolute(boost::filesystem::path(mod->config.sqlite_db_file_path));
+    boost::filesystem::path database_directory = sqlite_db_path.parent_path();
+    if (!boost::filesystem::exists(database_directory)) {
+        boost::filesystem::create_directories(database_directory);
+    }
+
+    int ret = sqlite3_open(sqlite_db_path.c_str(), &this->db);
+
+    if (ret != SQLITE_OK) {
+        EVLOG(error) << "Error opening PersistentStore database '" << sqlite_db_path << "': " << sqlite3_errmsg(db);
```

Comment:
There is this combined EVLOG_AND_THROW macro, although we have no guidelines on when to use it